wasm/test_page/js/worker.js

```diff
@@ -0,0 +1,294 @@
+max-length-factor: 2.0
+skip-cost: true
+cpu-threads: 0
+quiet: true
+quiet-translation: true
+gemm-precision: int8shiftAll
+`;
+
+  const modelFile = `${rootURL}/${languagePair}/${modelRegistry[languagePair]["model"].name}`;
+  const shortlistFile = `${rootURL}/${languagePair}/${modelRegistry[languagePair]["lex"].name}`;
+  const vocabFiles = [`${rootURL}/${languagePair}/${modelRegistry[languagePair]["vocab"].name}`,
+                      `${rootURL}/${languagePair}/${modelRegistry[languagePair]["vocab"].name}`];
+
+  const uniqueVocabFiles = new Set(vocabFiles);
+  log(`modelFile: ${modelFile}\nshortlistFile: ${shortlistFile}\nNo. of unique vocabs: ${uniqueVocabFiles.size}`);
+  uniqueVocabFiles.forEach(item => log(`unique vocabFile: ${item}`));
+
+  // Download the files as buffers from the given urls
+  let start = Date.now();
+  const downloadedBuffers = await Promise.all([downloadAsArrayBuffer(modelFile), downloadAsArrayBuffer(shortlistFile)]);
+  const modelBuffer = downloadedBuffers[0];
+  const shortListBuffer = downloadedBuffers[1];
+
+  const downloadedVocabBuffers = [];
+  for (let item of uniqueVocabFiles.values()) {
+    downloadedVocabBuffers.push(await downloadAsArrayBuffer(item));
+  }
+  log(`Total Download time for all files of '${languagePair}': ${(Date.now() - start) / 1000} secs`);
+
+  // Construct AlignedMemory objects with downloaded buffers
+  let constructedAlignedMemories = await Promise.all([prepareAlignedMemoryFromBuffer(modelBuffer, 256),
+                                                      prepareAlignedMemoryFromBuffer(shortListBuffer, 64)]);
+  let alignedModelMemory = constructedAlignedMemories[0];
+  let alignedShortlistMemory = constructedAlignedMemories[1];
+  let alignedVocabsMemoryList = new Module.AlignedMemoryList;
+  for(let item of downloadedVocabBuffers) {
+    let alignedMemory = await prepareAlignedMemoryFromBuffer(item, 64);
+    alignedVocabsMemoryList.push_back(alignedMemory);
+  }
+  for (let vocabs=0; vocabs < alignedVocabsMemoryList.size(); vocabs++) {
+    log(`Aligned vocab memory${vocabs+1} size: ${alignedVocabsMemoryList.get(vocabs).size()}`);
+  }
+  log(`Aligned model memory size: ${alignedModelMemory.size()}`);
+  log(`Aligned shortlist memory size: ${alignedShortlistMemory.size()}`);
+
+  log(`Translation Model config: ${modelConfig}`);
+  var translationModel = new Module.TranslationModel(modelConfig, alignedModelMemory, alignedShortlistMemory, alignedVocabsMemoryList);
+  translationModels.set(languagePair, translationModel);
+}
+
+const translate = (from, to, paragraphs) => {
+  // If none of the languages is English then perform translation with
+  // English as a pivot language.
+  if (from !== 'en' && to !== 'en') {
+    log(`Translating '${from}${to}' via pivoting: '${from}en' -> 'en${to}'`);
+    let translatedParagraphsInEnglish = translateInvolvingEnglish(from, 'en', paragraphs);
+    return translateInvolvingEnglish('en', to, translatedParagraphsInEnglish);
```

Comment:
This works only for plaintext, is not the recommended or intended way to do pivot translation. I'm unsure who communicated this is the solution for https://github.com/mozilla/bergamot-translator/issues/62#issue-958664129. Going forward with this for HTML will incur serious debt ahead -it won't work with links. But a targeted by the library pivot translation will be compatible with other algorithms like those which require TagTree.

https://github.com/browsermt/bergamot-translator/issues/212 is unsolved and will take more development time after which the proper solution will come about, if at all.

Reply:
We know that for the links/tags, the current setup wouldn't work. Once the API supports pivoting for html text, this page will be updated accordingly. This is quite clear from the https://github.com/mozilla/bergamot-translator/issues/62#issuecomment-932391160

Reply:
> I'm telling you this is not the intended way to do pivot translation

Once you have the pivot translation ready the intended way, I will change the wasm test page accordingly.

Reply:
> I'm telling you this is not the intended way to do pivot translation, demonstrating this as a reference has more downsides than upsides

@jerinphilip would you please point us to what's the `intended way to do pivot translation` for both html tags plain text then? I searched for implementation documentation but to no avail, and it would be helpful if you could provide those. 
Thanks!

Reply:
The intended way is you to call a new function mentioned in https://github.com/browsermt/bergamot-translator/issues/212#issue-960451083, where the issue is being tracked. You tell the C++ API, "Hey, I have these models which you can apply in series to translate.". This will give you a `Response` that is consistent, usable the same way as you would have in an English model, hence working with tags and what not, when the feature gets to you. 

This will shift the responsibility of resolving inconsistencies at alignment matrices, vocabulary mismatches, sentence misalignments to the C++ library. You will run into trouble doing this at JS at some point, and by then if this example code is used the efforts built on top of this example would go down the drain.

The feature is not implemented, hence still open. If I may ask, why are you under the impression that this has been implemented? I didn't communicate pivot translation is solved, but I believe it's my responsibility to point out pitfalls of the current approach when it's brought to my attention.

Reply:
> The intended way is you to call a new function mentioned in #212 (comment), where the issue is being tracked.
> The feature is not implemented, hence still open. 

@kpu Could you please confirm that until https://github.com/browsermt/bergamot-translator/issues/212 resolves, it is okay to perform translation b/w language pairs not involving English by using English as a pivot? I changed the wasm test page just to demonstrate that multiple models work in JS and used pivoting as a use case to demonstrate this. We are aware that the existing solution for pivoting works for only for the plain text and not for the html text. We would be happy to change things on our side once https://github.com/browsermt/bergamot-translator/issues/212 resolves. Please let me know if I missed something here. Thanks. 👍

Reply:
We do have systems lying around for Kazakh-English with Russian as a pivot (which have not been optimized for Bergamot).  

There will be API changes in the future so we can get word alignment working in pivot translation.  It will probably be done on C++ side.  As long as you accept that your code will have to change in the future to accommodate that (and Edinburgh is unlikely to be helpful with this since it's JavaScript), I have no problem with you putting out something that works for the time being.